util/analyze/_utils.py

```diff
@@ -0,0 +1,20 @@
+from ._types import *
+
+
+def foreach_bench(analysis_f, *logs):
+    '''
+    Repeats `analysis_f` for each benchmark in `logs`.
+    Also runs it for the entire thing.
+
+    Returns:
+        A dictionary containing the per-benchmark results.
+        The keys are the benchmark names.
+        The run for the entire thing has a key of 'Total'
+    '''
+
+    benchmarks = zip(*[log.benchmarks for log in logs])
+
+    return {
+        'Total': analysis_f(*logs),
+        **{bench[0].name: analysis_f(*bench) for bench in benchmarks}
+    }
```

Comment:
This could be a cause of performance problems, this code calls the analysis function on all the logs in line 18, then subsequently calls it for each benchmark in line 19. This likely doubles runtime(you said that runtimes were long).  Would it be possible to combine the analysis of the benchmarks to form the analysis of the total?

Reply:
Most of the performance problem is in loading the log files. Actually running the analysis is fairly fast. However, you're absolutely right that this is a good opportunity for a performance improvement.

------

I just measured performance for a script that just loads the logs and does nothing with it against one that does some useful analysis. Just loading logs: `0m28.228s`. Calculating some stats on the logs after loading it: `0m36.344s`. So most of the time is spent loading the logfile, but it could still be reasonable to optimize this (the calculated stats aren't the most extreme either)

Reply:
It was easy enough to fix, so I went ahead and did so.